stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt

```diff
@@ -831,6 +854,54 @@ public class MessageListController(
         )
     }
 
+    /**
+     *  Changes the current [_mode] to be [Thread] with [ThreadState] and Loads thread data using ChatClient
+     *  directly. The data is observed by using [ThreadState].
+     *
+     *  The difference between [enterThreadMode] and [enterThreadSequential] is that the latter makes a call to a
+     *  [ChatClient] extension function which will return a [ThreadState] instance only once the API call had finished,
+     *  while the former calls a different function which returns  a [ThreadState] instance immediately after the API
+     *  request has fired, regardless of its completion state.
+     *
+     * @param parentMessage The message with the thread we want to observe.
+     */
+    private suspend fun enterThreadSequential(parentMessage: Message) {
+        val threadState = chatClient.awaitRepliesAsState(parentMessage.id, DEFAULT_MESSAGES_LIMIT)
+        val channelState = channelState.value ?: return
+
+        _messageActions.value = _messageActions.value + Reply(parentMessage)
+        _mode.value = MessageMode.MessageThread(parentMessage, threadState)
+
+        observeThreadMessagesState(
+            threadId = threadState.parentId,
+            messages = threadState.messages,
+            endOfOlderMessages = threadState.endOfOlderMessages,
+            reads = channelState.reads
+        )
+    }
+
+    /**
+     * Fetches the message with the given ID internally and then calls [lastLoadedThreadMessage].
+     *
+     * @param parentMessageId The ID of the message we are trying to fetch.
+     */
+    private suspend fun enterThreadSequential(parentMessageId: String) {
+        val result = chatClient.getMessageUsingCache(parentMessageId).await()
+
+        when (result) {
+            is Result.Success -> {
+                enterThreadSequential(result.value)
+            }
+            is Result.Failure -> {
+                val error = result.value
```

Comment:
This local variable is not needed, it can be accessed directly on the log message

Reply:
Removed the local variable